From: Simon McVittie <smcv@debian.org>
Date: Tue, 30 Aug 2022 12:32:44 +0100
Subject: app: Print a warning message about focal being EOL

---
 app/flatpak-main.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/app/flatpak-main.c b/app/flatpak-main.c
index 2be240b..a7d395a 100644
--- a/app/flatpak-main.c
+++ b/app/flatpak-main.c
@@ -677,6 +677,16 @@ flatpak_run (int      argc,
 
   __attribute__((cleanup (uninstall_polkit_agent))) gpointer polkit_agent = NULL;
 
+  if (isatty (2))
+    g_printerr ("========================================================================\n"
+                "Ubuntu 20.04 'focal' reached end-of-life for general support in May 2025.\n"
+                "\n"
+                "The packages in the Flatpak PPAs for Ubuntu 20.04 will not necessarily\n"
+                "be updated, even if there are security vulnerabilities. Please upgrade\n"
+                "to a newer LTS branch of Ubuntu.\n"
+                "========================================================================\n"
+                "\n");
+
   command = extract_command (&argc, argv, &command_name);
 
   if (!command->fn)
