From: Simon McVittie <smcv@debian.org>
Date: Mon, 20 Aug 2018 15:22:00 +0100
Subject: build: Install httpcache if installed-tests are enabled

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/flatpak/pull/1989
---
 tests/Makefile.am.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/Makefile.am.inc b/tests/Makefile.am.inc
index 1575c23..b902a79 100644
--- a/tests/Makefile.am.inc
+++ b/tests/Makefile.am.inc
@@ -109,7 +109,7 @@ dist_test_scripts = \
 	$(NULL)
 
 test_programs = testlibrary
-noinst_PROGRAMS += httpcache
+test_extra_programs = httpcache
 
 @VALGRIND_CHECK_RULES@
 VALGRIND_SUPPRESSIONS_FILES=tests/flatpak.supp tests/glib.supp
