From: Simon McVittie <smcv@debian.org>
Date: Mon, 19 Nov 2018 14:05:16 +0000
Subject: Fix a typo that broke installed-tests

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/flatpak/pull/2338
---
 tests/libtest.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/libtest.sh b/tests/libtest.sh
index 2ccd695..4a5e699 100644
--- a/tests/libtest.sh
+++ b/tests/libtest.sh
@@ -209,7 +209,7 @@ make_runtime () {
             if [ ! -d ${RUNTIME_REPO} ]; then
                 $(dirname $0)/make-test-runtime.sh ${RUNTIME_REPO} org.test.Platform "" > /dev/null
             fi
-        ) 200>${TEST_DATADIR}/runtime-repo-lock
+        ) 200>${TEST_DATA_DIR}/runtime-repo-lock
     fi
 
     if [ ! -d repos/${REPONAME} ]; then
