diff --git a/tests/test-summaries.sh b/tests/test-summaries.sh
index ee8766ed..34b8eacf 100755
--- a/tests/test-summaries.sh
+++ b/tests/test-summaries.sh
@@ -71,7 +71,7 @@ EOF
     cp $(dirname $0)/org.test.Hello.png ${DIR}/files/share/app-info/icons/flatpak/64x64/${APP_ID}.png
 
     $FLATPAK build-finish --command=hello.sh ${DIR} &> /dev/null
-    $FLATPAK build-export ${GPGARGS} --arch=$APPARCH --disable-sandbox  ${REPO} ${DIR} &> /dev/null
+    $FLATPAK build-export --no-update-summary ${GPGARGS} --arch=$APPARCH --disable-sandbox  ${REPO} ${DIR} &> /dev/null
     rm -rf ${DIR}
 }
 
@@ -160,6 +160,8 @@ for A in $ARCHES; do
         make_app org.app.App$I $A repos/test
     done
 
+    update_repo
+
     # Make sure we have no superfluous summary files
     verify_subsummaries repos/test
 done
@@ -168,13 +170,23 @@ set -x
 
 ok subsummary update generations
 
+echo OTHER_ARCH: $OTHER_ARCH
+flatpak repo --subsets repos/test
+
 ACTIVE_SUBSET=$(active_subset repos/test)
 ACTIVE_SUBSET_OTHER=$(active_subset_for_arch repos/test $OTHER_ARCH)
 
+echo ACTIVE_SUBSET: $ACTIVE_SUBSET
+echo ACTIVE_SUBSET_OTHER: $ACTIVE_SUBSET_OTHER
+
+# Ensure we have no initial cache
+rm -rf $FL_CACHE_DIR/summaries/*
+
 assert_not_has_file $FL_CACHE_DIR/summaries/test-repo-${ACTIVE_SUBSET}.sub
 assert_not_has_file $FL_CACHE_DIR/summaries/test-repo-${ACTIVE_SUBSET_OTHER}.sub
 
 httpd_clear_log
+# Prime cache for default arch
 $FLATPAK $U remote-ls test-repo  > /dev/null
 
 assert_has_file $FL_CACHE_DIR/summaries/test-repo-${ARCH}-${ACTIVE_SUBSET}.sub
@@ -185,6 +197,8 @@ assert_file_has_content httpd-log summaries/${ACTIVE_SUBSET}.gz
 httpd_clear_log
 $FLATPAK $U remote-ls test-repo --arch=$OTHER_ARCH  > /dev/null
 
+ls -l $FL_CACHE_DIR/summaries/
+
 assert_has_file $FL_CACHE_DIR/summaries/test-repo-${ARCH}-${ACTIVE_SUBSET}.sub
 assert_has_file $FL_CACHE_DIR/summaries/test-repo-${OTHER_ARCH}-${ACTIVE_SUBSET_OTHER}.sub
 # We downloaded the full summary (not delta)
