From: Simon McVittie <smcv@debian.org>
Date: Tue, 15 Jan 2019 18:43:03 +0000
Subject: Install environment generator as an executable file

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/flatpak/pull/2596
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index fd3ee65..a4a347c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -153,7 +153,7 @@ flatpak.env: env.d/flatpak.env.in
 		-e "s|\@sysconfdir\@|$(sysconfdir)|" $< > $@
 
 envgendir = $(systemduserenvgendir)
-envgen_DATA = env.d/60-flatpak
+envgen_SCRIPTS = env.d/60-flatpak
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = flatpak.pc
