From: Simon McVittie <smcv@debian.org>
Date: Wed, 14 Feb 2018 14:28:53 +0000
Subject: Only require FUSE if we're still building the document portal

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 5e30096..1905236 100644
--- a/configure.ac
+++ b/configure.ac
@@ -243,7 +243,8 @@ AM_CONDITIONAL(ENABLE_DOCUMENT_PORTAL, test "x$enable_document_portal" = "xyes")
 
 PKG_CHECK_MODULES(OSTREE, [ostree-1 >= $OSTREE_REQS])
 
-PKG_CHECK_MODULES(FUSE, [fuse])
+AS_IF([test "x$enable_document_portal" = "xyes"],
+      [PKG_CHECK_MODULES(FUSE, [fuse])])
 
 PKG_CHECK_MODULES(JSON, [json-glib-1.0])
 
