From: Alexander Larsson <alexl@redhat.com>
Date: Thu, 1 Sep 2016 11:44:21 +0200
Subject: update: Don't stop the entire update if one app fails

Origin: upstream, 0.6.10, commit:a5536d0420df8a537ab8327319a431127b0ebed7
---
 app/flatpak-builtins-update.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/app/flatpak-builtins-update.c b/app/flatpak-builtins-update.c
index 77de3d5..0c5eb9b 100644
--- a/app/flatpak-builtins-update.c
+++ b/app/flatpak-builtins-update.c
@@ -109,15 +109,9 @@ do_update (FlatpakDir  * dir,
     {
       if (g_error_matches (update_error, FLATPAK_ERROR,
                            FLATPAK_ERROR_ALREADY_INSTALLED))
-        {
-          g_print (_("No updates.\n"));
-        }
+        g_print (_("No updates.\n"));
       else
-        {
-          g_propagate_error (error, update_error);
-          update_error = NULL;
-          return FALSE;
-        }
+        g_printerr ("Error updating: %s\n", update_error->message);
     }
   else
     {
@@ -172,7 +166,7 @@ do_update (FlatpakDir  * dir,
                                        FLATPAK_ERROR_ALREADY_INSTALLED))
                     g_print (_("No updates.\n"));
                   else
-                    g_printerr ("%s\n", local_error->message);
+                    g_printerr ("Error updating: %s\n", local_error->message);
                   g_clear_error (&local_error);
                 }
               else
